test/parallel/test-http-server-request-timeouts-mixed.js

```diff
@@ -58,74 +62,48 @@ function createClient(server) {
   return request;
 }
 
-server.listen(0, common.mustCall(() => {
+server.listen(0, common.mustCall(async () => {
```

Comment:
```suggestion
server.listen(0, common.mustCall(() => (async () => {
```

Reply:
Explain please

Reply:
If we wrap the async into a separate function, we can ensure that the promise does settle thanks to the `.then(common.mustCall)` part (if it never settles, the `.then` handler will never get called which would trigger an assertion).

Reply:
I tried the suggestion but its failing the tests and I didn't understand what you are intending. Sorry, I am new here. So, could you please commit the changes?

Reply:
> could you please commit the changes?

Done in https://github.com/nodejs/node/pull/43470/commits/bbc053875c7e8673d8bfbe50471a7f99ded889e3.